var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -81,6 +85,7 @@ class PyDask(PythonPackage):
     depends_on('py-numpy@1.11.0:',      type=('build', 'run'), when='@0.17.3: +dataframe')
     depends_on('py-numpy@1.13.0:',      type=('build', 'run'), when='@1.2.1: +dataframe')
     depends_on('py-numpy@1.15.1:',      type=('build', 'run'), when='@2020.12.0: +dataframe')
+    depends_on('py-numpy@1.16.0:',      type=('build', 'run'), when='@2021.4.0: +dataframe')
```

Comment:
https://github.com/dask/dask/commit/2598af25fd2ba3dd11482194928b48efb327f637
```suggestion
    depends_on('py-numpy@1.16.0:',      type=('build', 'run'), when='@2021.3.1: +dataframe')
```
Also above:
```diff
-    depends_on('py-numpy@1.10.4:',      type=('build', 'run'), when='+dataframe')
+    depends_on('py-numpy',              type=('build', 'run'), when='@:0.17.1 +dataframe')
+    depends_on('py-numpy@1.10.4:',      type=('build', 'run'), when='@0.17.2: +dataframe')
```
And below:
```diff
-    depends_on('py-toolz',              type=('build', 'run'), when='+dataframe')
+    depends_on('py-toolz',              type=('build', 'run'), when='@:0.6.1 +dataframe')
     depends_on('py-toolz@0.7.2:',       type=('build', 'run'), when='@0.7.0: +dataframe')
     depends_on('py-toolz@0.7.3:',       type=('build', 'run'), when='@0.14.1: +dataframe')
-    depends_on('py-toolz@0.8.2:',       type=('build', 'run'), when='@2.13.0: +dataframe')
+    # The dependency on py-toolz is non-optional starting version 2021.3.1
+    depends_on('py-toolz@0.8.2:',       type=('build', 'run'), when='@2.13.0:2021.3.0 +dataframe')
 
-    depends_on('py-partd',              type=('build', 'run'), when='+dataframe')
     depends_on('py-partd@0.3.2:',       type=('build', 'run'), when='@0.6.0: +dataframe')
     depends_on('py-partd@0.3.3:',       type=('build', 'run'), when='@0.9.0: +dataframe')
     depends_on('py-partd@0.3.5:',       type=('build', 'run'), when='@0.10.2: +dataframe')
     depends_on('py-partd@0.3.7:',       type=('build', 'run'), when='@0.13.0: +dataframe')
     depends_on('py-partd@0.3.8:',       type=('build', 'run'), when='@0.15.0: +dataframe')
-    depends_on('py-partd@0.3.10:',      type=('build', 'run'), when='@2.0.0: +dataframe')
+    # The dependency on py-partd is non-optional starting version 2021.3.1
+    depends_on('py-partd@0.3.10:',      type=('build', 'run'), when='@2.0.0:2021.3.0 +dataframe')
 
     depends_on('py-cloudpickle@0.2.1:', type=('build', 'run'), when='@0.8.2:2.6.0 +dataframe')
 
     depends_on('py-fsspec@0.3.3:',      type=('build', 'run'), when='@2.2.0: +dataframe')
     depends_on('py-fsspec@0.5.1:',      type=('build', 'run'), when='@2.5.0: +dataframe')
-    depends_on('py-fsspec@0.6.0:',      type=('build', 'run'), when='@2.8.0: +dataframe')
+    # The dependency on py-fsspec is non-optional starting version 2021.3.1
+    depends_on('py-fsspec@0.6.0:',      type=('build', 'run'), when='@2.8.0::2021.3.0 +dataframe')
```